core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -226,11 +227,16 @@ public void start() throws IOException {
    */
   @Override
   public void stop() throws IOException {
+    // Use shutdownNow because HeartbeatThreads never finish until they are interrupted
+    getExecutorService().shutdownNow();
```

Comment:
I've added comment. What happened during the test is that: the heartbeat and just keep holding the lock (keep retry) due to the following code block which prevents FileSystemMaster.close from finishing.

`  public synchronized Set<Long> getPinList() throws ConnectionFailedException, IOException {
    return retryRPC(new RpcCallable<Set<Long>>() {
      @Override
      public Set<Long> call() throws TException {
        return mClient.getPinIdList();
      }
    });
  }`